octodns/provider/yaml.py

```diff
@@ -248,10 +268,10 @@ def _apply(self, plan):
         self._do_apply(desired, data)
 
     def _do_apply(self, desired, data):
-        filename = join(self.directory, f'{desired.name}yaml')
+        filename = join(self.directory, f'{desired.decoded_name}yaml')
```

Comment:
Going forward the written filenames will use utf-8, decoded from idna if needed.